packages/data-table-server/src/middleware/attachDataTableFromPreviewToContext.ts

```diff
@@ -4,24 +4,18 @@
  */
 
 import { NextFunction, Request, Response } from 'express';
-import { getAjv } from '@tupaia/tsutils';
-import { generateId } from '@tupaia/database';
+import { ajvValidate } from '@tupaia/tsutils';
 import { DataTablePreviewRequestSchema } from '@tupaia/types';
 import type { DataTablePreviewRequest } from '@tupaia/types';
 
 import { getDataTableService } from '../dataTableService';
 import { validatePermissions } from './helpers';
 
-const validateDataTableFields = (dataTableFields: any): DataTablePreviewRequest => {
-  const ajv = getAjv();
-  const dataTableValidator = ajv.compile(DataTablePreviewRequestSchema);
-  // Add random id to meet validation requirement
-  const validate = dataTableValidator({ id: generateId(), ...dataTableFields });
-  if (!validate && dataTableValidator.errors) {
-    throw new Error(dataTableValidator.errors[0].message);
-  }
-
-  return dataTableFields;
+const validateDataTableFields = (dataTableFields: unknown) => {
+  const { properties: schemaProperties } = DataTablePreviewRequestSchema;
+  const { id, ...propertiesWithoutId } = schemaProperties;
+  const schemaWithoutId = { ...DataTablePreviewRequestSchema, properties: propertiesWithoutId };
+  return ajvValidate<Omit<DataTablePreviewRequest, 'id'>>(schemaWithoutId, dataTableFields);
```

Comment:
Felt the previous dance around of adding a fake id to get past the validator was slightly more awkward than stripping the id from the schema...

Reply:
I'm not sure this is possible. A json schema can represent A | B, with anyOf: https://json-schema.org/understanding-json-schema/reference/combining. The id property may not exist top level.

Reply:
> I was going to suggest in the general case where if want to CREATE/EDIT a model and we don't have an id, we should update our generation script to add WithoutId variants of each db model e.g. EntityWithoutId = Omit<Entity, 'id'>. So we would get that schema for free and across all models.

I like that, I can see those types coming in handy pretty often :+1: 

But you're also right that this request type is just wrong, I'll fix that now